src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -1952,4 +1960,127 @@ private string ReplaceLocalhostWithContainerHost(string value)
+                patch = CreatePatch(rs, obj => obj.Spec.Replicas = 0);
+                await kubernetesService.PatchAsync(rs, patch, cancellationToken).ConfigureAwait(false);
+                break;
+            default:
+                throw new InvalidOperationException($"Unexpected resource type: {matchingResource.DcpResource.GetType().FullName}");
+        }
+    }
+
+    private AppResource GetMatchingResource(string resourceName)
+    {
+        var matchingResource = _appResources
+            .Where(r => r.DcpResource is not Service)
+            .SingleOrDefault(r => string.Equals(r.DcpResource.Metadata.Name, resourceName, StringComparisons.ResourceName));
+        if (matchingResource == null)
+        {
+            throw new InvalidOperationException($"Resource '{resourceName}' not found.");
+        }
+
+        return matchingResource;
+    }
+
+    internal async Task StartResourceAsync(string resourceName, CancellationToken cancellationToken)
+    {
+        var matchingResource = GetMatchingResource(resourceName);
+
+        switch (matchingResource.DcpResource)
+        {
+            case Container c:
+                await StartExecutableOrContainerAsync(c).ConfigureAwait(false);
+                break;
+            case Executable e:
+                await StartExecutableOrContainerAsync(e).ConfigureAwait(false);
+                break;
+            case ExecutableReplicaSet rs:
+                var replicas = matchingResource.ModelResource.GetReplicaCount();
+                var patch = CreatePatch(rs, obj => obj.Spec.Replicas = replicas);
+
+                await kubernetesService.PatchAsync(rs, patch, cancellationToken).ConfigureAwait(false);
+                break;
+            default:
+                throw new InvalidOperationException($"Unexpected resource type: {matchingResource.DcpResource.GetType().FullName}");
+        }
+
+        async Task StartExecutableOrContainerAsync<T>(T resource) where T : CustomResource
+        {
+            var resourceNotFound = false;
+            try
+            {
+                await kubernetesService.DeleteAsync<T>(resource.Metadata.Name, cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+            catch (HttpOperationException ex) when (ex.Response.StatusCode == System.Net.HttpStatusCode.NotFound)
+            {
+                // No-op if the resource wasn't found.
+                // This could happen in a race condition, e.g. double clicking start button.
+                resourceNotFound = true;
+            }
+
+            // Ensure resource is deleted. DeleteAsync returns before the resource is completely deleted so we must poll
+            // to discover when it is safe to recreate the resource. This is required because the resources share the same name.
+            if (!resourceNotFound)
+            {
+                // Limit polling to 5 attempts to avoid hanging with an infinite loop.
+                for (var i = 0; i < 5; i++)
```

Comment:
Why don't you use polly 😄

Reply:
I've used polly zero times and I've used for loops a million times 😋

Next time I make start/stop/restart changes in this area I'll try it out.

Reply:
It's in this same file

Reply:
Since the build is failing anyway, changed to polly.

Observation: I haven't used polly before, but should the polly options and pipelines be cached? They're created every run right now. Might not be necessary.